stan/math/prim/meta/possibly_sum.hpp

```diff
@@ -0,0 +1,37 @@
+#ifndef STAN_MATH_PRIM_META_POSSIBLY_SUM_HPP
+#define STAN_MATH_PRIM_META_POSSIBLY_SUM_HPP
+
+#include <stan/math/prim/fun/sum.hpp>
+#include <stan/math/prim/meta/require_helpers.hpp>
+
+namespace stan {
+namespace math {
+
+/**
+ * Conditionally sum the input at compile time.
+ * @tparam CondSum A struct with a static boolean member `value` which if true
+ *  will allow the input value to be summed
+ * @tparam T A scalar, Eigen type, or standard vector with inner scalar type.
+ * @param x The value to be summed.
+ */
+template <typename CondSum, typename T, require_t<CondSum>* = nullptr>
```

Comment:
No comments, just here to say that this is neat and I'm looking forward to having it in